eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -185,10 +191,14 @@
     private final EndpointUtils.ServiceUrlRandomizer urlRandomizer;
     private final Provider<BackupRegistry> backupRegistryProvider;
     private final ApacheHttpClient4 discoveryApacheClient;
-    private EurekaHttpClient eurekaHttpClient;
-    private EurekaHttpClientFactory eurekaHttpClientFactory;
     private EurekaJerseyClient discoveryJerseyClient;
 
+    private EurekaHttpClient registrationClient;
+    private EurekaHttpClientFactory registrationClientFactory;
+
+    private EurekaHttpClient queryClient;
+    private EurekaHttpClientFactory queryClientFactory;
```

Comment:
We still could. This change was made as the two (reg and query) clients need different resolvers that may internally be backed by threadpools, and given that we give the option of disabling reg and query, it makes sense to separate them so init can be skipped.